packages/form-state-manager/src/utils/use-subscription.ts

```diff
@@ -157,33 +122,49 @@ const useSubscription = ({
     if (validate) {
       const error = fieldLevelValidator(validate, sanitizedValue, formOptions().values, formOptions);
       if (isPromise(error)) {
-        setState((prevState) => ({ ...prevState, meta: { ...prevState.meta, validating: true } }));
+        setState((prevState: FieldState) => ({ ...prevState, meta: { ...prevState.meta, validating: true } }));
         const asyncError = error as Promise<string | undefined>;
         asyncError.then(() => handleError(true)).catch((error) => handleError(false, error));
       } else {
         const syncError = error as string | undefined;
         if (error) {
           handleError(false, syncError);
-        } else if (state.meta.valid === false) {
+        } else if (formOptions().getFieldState(name)?.meta.valid === false) {
           handleError(true);
         }
       }
     }
 
-    setState((prevState) => ({ ...prevState, value: getFieldValue(name) }));
+    setState((prevState: FieldState) => ({ ...prevState, value: getFieldValue(name) }));
   };
 
-  const valueToReturn = state.value;
+  const valueToReturn = formOptions().getFieldValue(name);
+
+  const { current: initValue } = useRef(initialValue || get(initialValues, name));
 
   useEffect(() => {
```

Comment:
We should probably move the registration code to top and do it before the rest of the code

```jsx
useState(() => {  // lazy-init... we can do it differently
    registerField(...)
})
```

This will allow us to do `const state = formOptions().getFieldState(name)` right after it and have the state synced with the global state... and we don't have to use `useRef` or store here anything anymore.